### DISCUSSION TITLE
Why does this script location not exist?

### DISCUSSION THREAD

**Discussion by alkene5** [johnlindquist/kit#1371] [Q&A]
I use the following method call, found that there is no image in the clipboard, and use the keyword trigger can get the image
```js
await run(kitPath("main", "clipboard-history.js"))
```
<img width="773" alt="image" src="https://github.com/johnlindquist/kit/assets/61105068/198aaf56-201b-4c77-aeb0-7a271868f3f9">

<img width="768" alt="image" src="https://github.com/johnlindquist/kit/assets/61105068/e9aefaf6-4d39-4440-9456-83e7af7f411e">

**Comment by johnlindquist** [comment]
Because "reveal-script" is using `selectScript` which checks the `~/.kit/db/scripts.json`. The main Kit SDK scripts are not included in that file.

https://github.com/johnlindquist/kit/blob/main/src/cli/reveal-script.ts

`selectScript` is used in many of the Kit SDK scripts because it will pop open the script selection dialog, but in this case, the filePath is being passed into it from the main menu causing the confusion. I just refactored it locally to this:

```js
let filePath = ""
if (typeof args[0] === "string") {
  filePath = args[0]
} else {
  let script = await selectScript(
    `Which script do you want to reveal?`
  )
  filePath = script.filePath
}
await revealFile(filePath)
```

So if an arg is passed in, just skip the menu. I'll get it committed later

**Reply by alkene5** [reply]
How do I set global shortcuts for default scripts? like 'clipboard-history.js', I'm doing that now, but it doesn't work very well
```js
// Name: Proxy
// Shortcut: cmd shift v

await run(kitPath("main", "clipboard-history.js"))
```

**Reply by johnlindquist** [reply]
What do you mean by "doesn't work very well"? Is something broken? Is it slow?

That's the proper way to do it, so if there's something wrong, I'll try to fix it.

You can also use:

```js

let history = await getClipboardHistory()
dev(history)
```

if you want to set up your own clipboard history tool

**Reply by alkene5** [reply]
There is no picture displayed when triggered by a shortcut key, When triggered by a keyword 'c ', everything works
1. Keyword
<img width="771" alt="image" src="https://github.com/johnlindquist/kit/assets/61105068/aaf5bd03-3a1e-49a7-a3fc-d59b7b74118e">
2. Shortcut
<img width="787" alt="image" src="https://github.com/johnlindquist/kit/assets/61105068/9b7e674a-e8ae-435d-8b1d-23319406dbb1">

**Reply by alkene5** [reply]
`await getClipboardHistory()` It works
